src/ipa_pc_de/mod.rs

```diff
@@ -261,7 +255,7 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
         Ok(Some(check_poly))
     }
 
-    /// The succinct portion of verifying a multi-poly single-point opening proof.
+    /// The succinct verifier a multi-poly single-point opening proof.
     /// If successful, returns the (recomputed) reduction challenge.
     pub fn succinct_single_point_multi_poly_verify<'a>(
         vk: &VerifierKey<G>,
```

Comment:
Why do we need to define the struct LabeledCommitment  and we do not use the struct Commitment?
Why labeled_commitments and values are not just vectors.

Reply:
In general labeled structures are needed to be agnostic from the order of elements. We are not receiving these parameters in synced order, so we use labels to restore relations